seed/serialize.py

```diff
@@ -26,6 +27,73 @@
     'RELEASE': study_pb2.Study.Channel.STABLE
 }
 
+def version_to_int_array(version_str):
+    version_list = []
+    if version_str is None:
+        return version_list
+
+    parts = version_str.split('.')
+    for part in parts:
+        if part == '*':
+            version_list.append(part)
+            break
+        version_list.append(int(part))
+
+    return version_list
+
+def compare_versions(version1, version2):
+    min_len = None
+    if not version1:
+        min_len = 0
+    elif version1[-1] == '*':
+        version1 = version1[:-1]
+        min_len = len(version1)
+
+    if not version2:
+        min_len = 0
+    elif version2[-1] == '*':
+        version2 = version2[:-1]
+        if min_len is not None:
+            min_len = min(min_len, len(version2))
+        else:
+            min_len = len(version2)
+
+    if min_len is not None:
+        version1 = version1[:min_len]
+        version2 = version2[:min_len]
+
+    if version1 > version2:
+        return 1
+    elif version1 < version2:
+        return -1
+    else:
+        return 0
+
+def test_version_comparison():
+    # //base/version_unittest.cc VersionTest.CompareToWildcardString
+    test_cases = [
+        ["1.0", "1.*", 0],
+        ["1.0", "0.*", 1],
+        ["1.0", "2.*", -1],
+        ["1.2.3", "1.2.3.*", 0],
+        ["10.0", "1.0.*", 1],
+        ["1.0", "3.0.*", -1],
+        ["1.4", "1.3.0.*", 1],
+        ["1.3.9", "1.3.*", 0],
+        ["1.4.1", "1.3.*", 1],
+        ["1.3", "1.4.5.*", -1],
+        ["1.5", "1.4.5.*", 1],
+        ["1.3.9", "1.3.*", 0],
+        ["1.2.0.0.0.0", "1.2.*", 0],
+        [None, None, 0],
+        [None, "1", 0],
+        ["1", None, 0],
+    ]
+    for test_case in test_cases:
+        version1 = version_to_int_array(test_case[0])
+        version2 = version_to_int_array(test_case[1])
+        assert compare_versions(version1, version2) == test_case[2]
+
```

Comment:
this is pretty dirty (having tests and logic in the same file), but we will remove this soon entirely.